src/js/components/DateInput/DateInput.js

```diff
@@ -185,7 +188,7 @@ const DateInput = forwardRef(
           id={id}
           dropProps={{ align: { top: 'bottom', left: 'left' }, ...dropProps }}
           dropContent={calendar}
-          icon={<CalendarIcon size={iconSize} />}
+          icon={icon || <CalendarIcon size={iconSize} />}
```

Comment:
We should still allow the icon to be set through input props.
```suggestion
          icon={icon || MaskedInputIcon || <CalendarIcon size={iconSize} />}
```